### ISSUE TITLE
Question about traffic_light_state in Motion dataset

### ISSUE THREAD

**Issue by SwagJ** [waymo-research/waymo-open-dataset#442]
Hi, 

I have a question about the variable traffic_**light_state/future/valid** in tf.example. For testing and validation set, this information is marked as all 0, while not for training set. Is this true? If so, I think it is a bit of strange since the future state of traffic light state might influence trajectories heavily.

**Comment by scott-ettinger** [comment]
Hi, 
The traffic_light_state/future/valid tensors are populated in both the training and validation dataset splits (could you check again and let me know if you do not see any data populated in the validation set?  I do see data there).  The test set does not contain any future data of any kind because to be realistic, models cannot take any future data as inputs (models must be causal). 

We provide the future traffic light data in the training and validation splits for other research purposes, but for prediction models, that data cannot be used as inputs.

**Comment by SwagJ** [comment]
Hi @scott-ettinger,

Thank you for your reply. I think the past of future traffic state's data shape is wrong in the tf_example specs, should be 10, 80 instead of 1, right? Also, for roadgraphs, is there any any api functions built for getting information of map_proto from tf_example data, such as lane neighbor, lane centers and etc?

**Comment by scott-ettinger** [comment]
Thank you for pointing this out.  We will fix the incorrect shapes on the website.  
Currently there is not a way to get the lane neighbors and boundaries from the tf.Examples.  You would need to write code to convert the data in the Scenario protos into tensors.  We will consider adding this data to the tf.Example format in future releases.

**Comment by SwagJ** [comment]
Hi,

Thank you for your reply. I will decompose Scenario data myself.
